server/service/src/asset/parse.rs

```diff
@@ -0,0 +1,231 @@
+use chrono::Utc;
+use repository::{
+    asset::{Asset, AssetFilter, AssetRepository},
+    asset_catalogue_item::{AssetCatalogueItemFilter, AssetCatalogueItemRepository},
+    EqualFilter, RepositoryError, StringFilter,
+};
+use util::{GS1ParseError, GS1};
+
+use crate::service_provider::ServiceContext;
+
+#[derive(Debug)]
+pub enum ScannedDataParseError {
+    ParseError,
+    MissingPartNumber,
+    MissingSerialNumber,
+    NotFound,
+    DatabaseError(RepositoryError),
+}
+
+impl From<RepositoryError> for ScannedDataParseError {
+    fn from(error: RepositoryError) -> Self {
+        ScannedDataParseError::DatabaseError(error)
+    }
+}
+
+fn lookup_asset_by_id(ctx: &ServiceContext, id: &str) -> Result<Asset, ScannedDataParseError> {
+    let repository = AssetRepository::new(&ctx.connection);
+
+    let mut result =
+        repository.query_by_filter(AssetFilter::new().id(EqualFilter::equal_to(id)))?;
```

Comment:
NIT: but any reason to use query_by_filter rather than just `find_one_by_id` off the row repository?

Reply:
This won't find deleted items, but I guess either way is ok? If its been deleted what should be the behaviour?